build.sbt

```diff
@@ -1,21 +1,27 @@
-lazy val root = (project in file(".")).
-  settings(
-    inThisBuild(List(
+lazy val root = (project in file(".")).settings(
+  inThisBuild(
+    List(
       organization := "com.example",
       scalaVersion := "2.13.3",
-      version      := "0.1.0-SNAPSHOT")),
-    name := "LibreCaptcha",
-
-    libraryDependencies += "com.sksamuel.scrimage" % "scrimage-core" % "4.0.5",
-
-    libraryDependencies += "com.sksamuel.scrimage" % "scrimage-filters" % "4.0.5",
-
-    libraryDependencies += "org.json4s" % "json4s-jackson_2.13" % "3.6.9"
-
+      version := "0.1.0-SNAPSHOT",
+      semanticdbEnabled := true,
+      semanticdbVersion := scalafixSemanticdb.revision,
+      scalafixScalaBinaryVersion := "2.13"
+    )
+  ),
+  name := "LibreCaptcha",
+  libraryDependencies += "com.sksamuel.scrimage" % "scrimage-core" % "4.0.5",
+  libraryDependencies += "com.sksamuel.scrimage" % "scrimage-filters" % "4.0.5",
+  libraryDependencies += "org.json4s" % "json4s-jackson_2.13" % "3.6.9"
 )
 
-unmanagedResourceDirectories in Compile += {baseDirectory.value / "lib"}
+unmanagedResourceDirectories in Compile += { baseDirectory.value / "lib" }
+scalacOptions ++= List(
+  "-Yrangepos",
+  "-Ywarn-unused"
+)
 javacOptions += "-g:none"
+scalafmtOnCompile := true
```

Comment:
If it formats on compile, wouldn't all PRs pass the CI check? Because the compilation in CI would format it and make everything appear to be correct?

Reply:
Only the formatter runs on compile, which essentially only takes care of spacing rules and structure.
The linter which runs during CI checks for more complex things like unused imports or variables etc.

Reply:
Also, since it only checks for spacing and structure, I thought it would be nice if it reformats the file before committing the changes. A standard template all contributors follow automatically.

Reply:
Ok, I have never used something like this. Let's see how it goes.